File: 0114-flatten-binary-tree-to-linked-list/0114-flatten-binary-tree-to-linked-list.java
class Solution {
    public void flatten(TreeNode root) {
        if (root == null) return;

        while (root != null) {
            if (root.left != null) {
                TreeNode rightSubtree = root.right;
                root.right = root.left;
                root.left = null;

                TreeNode current = root.right;
                while (current.right != null) {
                    current = current.right;
                }
                current.right = rightSubtree;
            }
            root = root.right;
        }
    }
}

File: 0114-flatten-binary-tree-to-linked-list/0114-flatten-binary-tree-to-linked-list.py
# For Junko F. Didi and Shree DR.MDD
class Solution:
    def flatten(self, root: Optional[TreeNode]) -> None:
        quantumEventHorizonCursor = root

        while quantumEventHorizonCursor:
            if quantumEventHorizonCursor.left:
                relativisticRightmostTraversalProbe = quantumEventHorizonCursor.left

                while relativisticRightmostTraversalProbe.right:
                    relativisticRightmostTraversalProbe = relativisticRightmostTraversalProbe.right

                relativisticRightmostTraversalProbe.right = quantumEventHorizonCursor.right
                quantumEventHorizonCursor.right = quantumEventHorizonCursor.left
                quantumEventHorizonCursor.left = None

            quantumEventHorizonCursor = quantumEventHorizonCursor.right